### PR TITLE
10195 download audio from audiocom

### LINKED ISSUE #10195: Sign in/ sign out to audio.com
### Description

Sign in / account

Sign-in flow. Same as AU3 (Create cloud account & Sign into cloud dialogs with Google/FB sign in options)
'My accounts' section in the 'Home' tab replaces the 'Cloud' option in preferences from AU3. Check the MuseScore Studio implementation. We exactly want this.

### LINKED ISSUE #10217: audio.com error messages handling
### Description

Notifications about network errors, sync failure, duplicated projects/audio names, etc

### LINKED ISSUE #10754: Crash when saving 2nd cloud project in a row
1. Sign in to Audio.com
2. In the cloud project tab, click an existing project
3. Once the project is open, save it
4. Close it
5. Open another project
6. Try to save it
7. Observe: a crash

https://github.com/user-attachments/assets/f1d9da89-c622-4f33-8e47-99ac924900b5

### PR DESCRIPTION
**Pull request by embarc-gabriel**
Resolves: #10195, https://github.com/audacity/audacity/issues/10217

We can now download an audio file from audio.com and open it on a separate project.

I'll ignore the rabbit :rabbit: now because it has already reviewed the code and the changes were just fixes for its comments. 

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

QA:

- [ ] Autobot test cases have been run
- [x] Check the new change for the thumbnails(preview) for audio and cloud projects
- [x] Test thumbnails on list and grid view
- [x] Click on the list should open the project/audio?
- [x] Context menu for audio
- [x] Open an audio when there is already a project opened should open in a new window
- [ ] Test corner cases to check #10217

@coderabbitai ignore

### CODE REVIEW

File: src/au3cloud/internal/au3audiocomtypeconv.cpp [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    projectList.meta.total = paginatedResponse.Pagination.TotalCount;
+    projectList.meta.batches = paginatedResponse.Pagination.PagesCount;
+    projectList.meta.thisBatchNumber = paginatedResponse.Pagination.CurrentPage;
+    projectList.meta.itemsPerBatch = paginatedResponse.Pagination.PageSize;
+
+    return projectList;
+}
+
+AudioList convertFromAu3CloudAudio(const audacity::cloud::audiocom::sync::PaginatedAudioResponse& paginatedResponse,
+                                   const muse::io::path_t& thumnailCacheDir)
```
[the remark this answers is not in the document]
Comment (embarc-gabriel) [reply src/au3cloud/internal/au3audiocomtypeconv.cpp]: Duplicate

File: src/project/qml/Audacity/Project/internal/ProjectsPage/ProjectListItem.qml [hunk 2]
```diff
@@ -79,80 +90,48 @@ ListItemBlank {
...
+                sourceComponent: CloudProjectIndicatorButton {
+                    id: cloudIndicator
 
-                        Layout.alignment: Qt.AlignTrailing | Qt.AlignVCenter
+                    Layout.alignment: Qt.AlignTrailing | Qt.AlignVCenter
 
-                        isProgress: false //cloudProjectStatusWatcher.isProgress
-                        isDownloadedAndUpToDate: true //cloudProjectStatusWatcher.isDownloadedAndUpToDate
+                    isProgress: false //cloudProjectStatusWatcher.isProgress
+                    isDownloadedAndUpToDate: true //cloudProjectStatusWatcher.isDownloadedAndUpToDate
```
[the remark this answers is not in the document]
Comment (embarc-gabriel) [reply src/project/qml/Audacity/Project/internal/ProjectsPage/ProjectListItem.qml:126]: It will be implemented later.

File: src/project/internal/projectactionscontroller.cpp [hunk 3]
```diff
@@ -1054,6 +1071,44 @@ void ProjectActionsController::shareAudio()
...
+    if (audioId.empty()) {
+        return;
+    }
+
+    auto progress = audioComService()->openAudioFile(audioId);
+    if (!progress) {
+        return;
+    }
+
+    progress->finished().onReceive(this, [this, audioId](const ProgressResult& result) {
```
Comment (kryksyh) [inline src/project/internal/projectactionscontroller.cpp]: we should do a cleanup and delete the temporary after it is imported or cancelled

Comment (kryksyh) [reply src/project/internal/projectactionscontroller.cpp]: if we fall into this branch, the file remains on the disk

```
        const auto project = globalContext()->currentProject();
        if (project) {
            QStringList args;
            args << "--session-type" << "start-with-new" << result.val.toQString();
            multiwindowsProvider()->openNewWindow(args);
            return;
        }
```

**Review by kryksyh — APPROVED** [review]
